tree/ntuplebrowse/v7/test/ntuple_browse.cxx

```diff
@@ -0,0 +1,500 @@
+#include <ROOT/RBrowseVisitor.hxx>
+#include <ROOT/RNTuple.hxx>
+#include <ROOT/RNTupleBrowser.hxx>
+#include <ROOT/RNTupleModel.hxx>
+
+#include <TDirectory.h>
+#include <TFile.h>
+
+#include "gtest/gtest.h"
+
+#include <memory>
+#include <sstream>
+#include <typeinfo>
+#include <vector>
+
+using RFolder = ROOT::Experimental::RNTupleBrowseFolder;
+using RNonFolder = ROOT::Experimental::RNTupleBrowseLeaf;
+using RNTupleBrowser = ROOT::Experimental::RNTupleBrowser;
+using RNTupleModel = ROOT::Experimental::RNTupleModel;
+using RNTupleReader = ROOT::Experimental::RNTupleReader;
+using RNTupleWriter = ROOT::Experimental::RNTupleWriter;
+
+/* This test cannot test the entire TBrwoser-support, it has no access to the TBrowser GUI.
+ * It only test certain functions, things it cannot test are:
+ * - The way the ntuple is displayed in TBrowser.
+ * - If histogram is displayed correctly.
+ * - It doesn't know what happens after TBrowser::Add() is called, and also doesn't know how often it was called.
+ */
+
+namespace {
+  /**
+   * An RAII wrapper around an open temporary file on disk. It cleans up the guarded file when the wrapper object
+   * goes out of scope.
+   */
+class FileRaii {
+private:
+   std::string fPath;
+   TFile *file;
+public:
+   FileRaii(const std::string &path) :
+      fPath(path),
+      file(TFile::Open(fPath.c_str(), "RECREATE"))
+      { }
+   FileRaii(const FileRaii&) = delete;
+   FileRaii& operator=(const FileRaii&) = delete;
+   TFile* GetFile() const { return file; }
+   ~FileRaii() {
+      file->Close();
+      std::remove(fPath.c_str());
+   }
+};
+} // anonymous namespace
+
+// helper function for tests
+std::string GetFileName(const TDirectory *directory) {
+   std::string fullPath(directory->GetPath());
+   return std::string(fullPath, 0, fullPath.find(".root") + 5);
+}
+
+TEST(RNTupleBrowse, Floattest)
+{
+   FileRaii fileGuard("test.root");
```

Comment:
One more minor comment on the naming: in order to prevent name clashes, please make the name more unique, e.g. `"test_ntuple_browse_float.root"`. Similar for the other tests.